hudi-common/src/main/java/org/apache/hudi/io/storage/HoodieHFileReader.java

```diff
@@ -236,6 +254,9 @@ public void close() {
     try {
       reader.close();
       reader = null;
+      if (keyScanner != null) {
```

Comment:
again, not sure of code paths. so, looks like close() does not require handling thread safety is it.

Reply:
Better to force thread safety here too.